src/config.nim

```diff
@@ -112,12 +126,15 @@ proc setCommandName*(s: string) =
 proc getChalkRuntime*(): ConfigState      = con4mRuntime.configState
 proc getValidationRuntime*(): ConfigState = con4mRuntime.validationState
 
-proc getKeySpec*(name: string): Option[KeySpec] =
-  if name in chalkConfig.keyspecs: return some(chalkConfig.keyspecs[name])
+proc getKeySpec*(name: string): Option[AttrScope] =
+  for k, v in getChalkSubsections("keyspec"):
```

Comment:
chalk has a lot of keys. no way to get these as a dict for O(1) lookup if the key is present?

Reply:
in docker util it uses:

https://github.com/crashappsec/chalk/pull/329/files#diff-53ca1218705d3afa936ec476df42cf4fcd3f6d02a0a99aad528a1778d9ec808dR110

```
  if key notin getContents(getChalkScope().getObject("keyspec")):
```